### PR TITLE
Add remote insights enabling

### PR DESCRIPTION
**Pull request by m-samiec**
#### Description
Added `SpatialExecServerCmd` console command with one command `StartInsights` to dynamically enable insights capturing.
Engine PR: https://github.com/improbableio/UnrealEngine/pull/932

Usage:
SpatialExecServerCmd < server > < command > < args >

Flow of events;
On execution the client validates that the command is a known valid command otherwise errors and returns.
The valid command is then sent via the client's player controller to the client's owning server.
The owning server parses the < server > argument to determine the destination server for the command.
- < server > can be "local" and will be executed directly on the client's owning server.
- < server > can also be the server worker name (eg. "UnrealWorker1398FEE345526D0B9EFC52BA7F5C094F").
- If < server > is not matched to an existing server, an error is output and the command is aborted.

Once the command is on the correct server, the individual command can be executed (the only valid command currently is `StartInsights`).
StartInsights passes <args> through to the Trace tool which parses the < args > in the same manner it would starting command line arguments.

Example usage:
SpatialExecServerCmd local StartInsights -trace=CustomChannel -tracefile=MyNewTrace 

If a trace is already active either via the command line launch arguments, or a previous call to "StartInsights", then the command is ignored.

#### Release note
Added

#### Primary reviewers
@improbable-danny @cmsmithio @martin-improbable

### CODE REVIEW

**Comment by cmsmithio** [comment]
Looks great Mike, any chance we will run into issues with log file name collisions if we turn this on for multiple servers?

**Review by cmsmithio — COMMENTED** [review]
just one comment/question posted

**Comment by cmsmithio** [comment]
> Looks great Mike, how hard do you think it would be to autogen the log names from the worker name?

Mike and I discussed this, sounds like it will autogenerate a log name in the case it is omitted.

**Comment by m-samiec** [comment]
> > Looks great Mike, how hard do you think it would be to autogen the log names from the worker name?
> 
> Mike and I discussed this, sounds like it will autogenerate a log name in the case it is omitted.

Yes in the case it's omitted (eg. "-tracefile=") it generates a filename from the date-time - FDateTime::Now().ToString(TEXT("%Y%m%d_%H%M%S.utrace"))

**Review by martin-improbable — CHANGES_REQUESTED** [review]
A few minor things, apart from that it looks very good!

File: SpatialGDK/Source/SpatialGDK/Private/Interop/DebugMetricsSystem.cpp [hunk 1]
```diff
@@ -51,29 +52,52 @@ void DebugMetricsSystem::ProcessOps(const TArray<Worker_Op>& Ops) const
...
-					Worker_CommandResponse Response = {};
-					Response.component_id = ComponentId;
-					Response.command_index = CommandIndex;
-					Response.schema_type = Schema_CreateCommandResponse();
+					Schema_Object* Payload = Schema_GetCommandRequestObject(CommandRequest.request.schema_type);
+					SpatialMetrics.OnExecServerCmdCommand(Payload);
+					bCommandHandled = true;
+					break;
+				}
+				}
```
[the remark this answers is not in the document]
Comment (m-samiec) [reply SpatialGDK/Source/SpatialGDK/Private/Interop/DebugMetricsSystem.cpp:81]: There's another commands on the server worker entity that get handled elsewhere - SpatialPlayerSpawner and the forward_spawn_player command

**Review by martin-improbable — APPROVED** [review]

File: SpatialGDK/Source/SpatialGDK/Private/Utils/SpatialMetrics.cpp [hunk 2]
```diff
@@ -4,13 +4,46 @@
...
+};
+
+const FString ServerCommandNames[static_cast<uint8>(EServerCommands::ServerCommandsCount) + 1] = { TEXT("StartInsights"), TEXT("Invalid") };
+
+const FString& ServerCommandsEnumToString(const EServerCommands Command)
+{
+	return ServerCommandNames[static_cast<uint8>(Command)];
+}
+
+const EServerCommands ServerCommandsStringToEnum(const FString& Command)
```
Comment (improbable-danny) [inline SpatialGDK/Source/SpatialGDK/Private/Utils/SpatialMetrics.cpp:33]: This functionality can be provided by UENUM but would require ServerCommandInvalid to be it's own value I think

Comment (improbable-danny) [reply SpatialGDK/Source/SpatialGDK/Private/Utils/SpatialMetrics.cpp:33]: Think I might be misremembering this..

Comment (improbable-danny) [reply SpatialGDK/Source/SpatialGDK/Private/Utils/SpatialMetrics.cpp:33]: Ah you can do EnumToString but StringToEnum is just a loop/cmp over EnumToString(i) = String

**Review by improbable-danny — APPROVED** [review]

**Pull request merged**